app/controllers/applets/ppe_collection_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Applets::PPECollectionController < ApplicationController
+  def index
+    if params[:hardhat_barcode]
+      return_hardhat
+    end
+  end
+
+  def return_hardhat
+    @hardhat = Tool.hardhats.find_by(barcode: params[:hardhat_barcode])
+    if @hardhat.nil?
+      redirect_to ppe_collection_path, alert: "Hardhat not found"
+    else
+      @checkout = @hardhat.checkouts.current.first unless @hardhat.checkouts.blank? || @hardhat.checkouts.current.blank?
+      if @hardhat.checkouts.blank? || @hardhat.checkouts.current.blank?
+        return redirect_to ppe_collection_path, alert: "Hardhat not checked out"
+      end
+      @checkout.checked_in_at = Time.zone.now
+      @checkout.save!
+
+      redirect_to ppe_collection_path, notice: "Hardhat #{params[:barcode]} successfully checked in."
```

Comment:
random suggestion: maybe also show the org it was checked out to and how many hard hats that org still has checked out.